controllers/front/callbackfail.php

```diff
@@ -188,11 +202,17 @@ public function postProcess()
 
         // Successful order exists, set its status to cancel
         $order = getOrderFromUniqueId($postData['shop_orderid']);
+
         if (Validate::isLoadedObject($order) and $transaction->ReservedAmount == 0) {
-            updatePaymentStatus($postData['transaction_id'], $postData['status']);
-            saveLastErrorMessage($postData['transaction_id'], $errorMessage);
-            $orderStatusCancelled = (int) Configuration::get('PS_OS_CANCELED');
-            setOrderStateIfNotExistInHistory($order, $orderStatusCancelled);
+            if ($isChildOrder) {
+                updatePaymentStatusForChildOrder($postData['transaction_id'], $postData['status']);
+                saveLastErrorMessageForChildOrder($postData['transaction_id'], $errorMessage);
+            } else {
+                updatePaymentStatus($postData['transaction_id'], $postData['status']);
+                saveLastErrorMessage($postData['transaction_id'], $errorMessage);
+                $orderStatusCancelled = (int) Configuration::get('PS_OS_CANCELED');
```

Comment:
Same as above

Reply:
@emicha, we make sure the cart object and its ID are valid before using them in the code. Here's how we do it:

```
            // Load the cart
            $cart = getCartFromUniqueId($shopOrderId);
            if (!Validate::isLoadedObject($cart)) {
                exit('Could not load cart - exiting');
            }
```
This check ensures that cart->id is always valid, which prevents any potential issues with casting.